utilities/license-header/check-license-header.sh

```diff
@@ -20,39 +20,26 @@
 ##
 ## ---------------------------------------------------------------------
 
-# check argument
+# check arguments
 if [[ -f $1 ]]; then
-  filepath=$1
+  file=$1
 else
-  echo "no file provided"
+  echo "No file provided!"
   exit 1
 fi
 
-# determine file name without full path
-filename=$(basename $filepath)
-
-# examine file type and determine comment symbol
-if [[ $filename == CMakeLists.txt || $filename == *.cmake || $filename == *.sh ]]; then
-  comment="##"
-elif [[ $filename == *.h || $filename == *.h.in || $filename == *.cc ]]; then
-  comment="\/\/" # note: escape the forward slashes
-elif [[ $fileext == *.rst ]]; then
-  comment=".."
+if [[ ! -z $2 ]]; then
```

Comment:
The `-n` option also checks for a non empty string.
```suggestion
if [[ -n $2 ]]; then
```

Reply:
Good catch! I adapted that everywhere.